### PR TITLE
Case views cleanup

### PR DESCRIPTION
**Pull request by dannyroberts**
This pull request makes all references to `case/all_cases` go through dbaccessors. For each case I did this in one of three ways:
- move the function to a db accessors file and clean up its interface
- use an existing db accessor that does the same thing (may use a different view)
- delete the code

In the process I also got rid of the `CaseQueryMixin`, which I felt was too direct/generic a wrapper for that specific view, which doesn't really let you refactor the implementation. I replaced all usages of it with calls to functions whose names specifically reflect what they're being used for, so it's easy to swap out the backend if desired.

**Note**: there is one behavioral change that I'm aware of and that is that as an artifact of switching some view calls to use `hqcase/types_by_domain` instead of `case/all_cases`, cases with missing/null/empty user ids are now no longer excluded from the result. This doesn't worry me, that's a pretty rare edge case (you can't configure an app to do that on the app builder), and it's unclear why you would want to exclude such cases anyway.

### CODE REVIEW

File: corehq/apps/api/resources/v0_1.py [hunk 1]
```diff
@@ -312,10 +316,15 @@ def obj_get_list(self, bundle, **kwargs):
...
-        return list(cases)
 
+        if include_closed:
+            case_ids = get_case_ids_in_domain(domain, type=case_type)
+        else:
+            case_ids = get_open_case_ids_in_domain(domain, type=case_type)
+
+        cases = imap(CommCareCase.wrap,
+                     iter_docs(CommCareCase.get_db(), case_ids))
+        return list(cases)
```
Comment (dannyroberts) [inline corehq/apps/api/resources/v0_1.py]: Does our case api really just read all of a domain's cases into memory!?

File: custom/_legacy/mvp/management/commands/mvp_force_update.py [hunk 2]
```diff
@@ -1,7 +1,9 @@
-from gevent import monkey;
+from gevent import monkey; monkey.patch_all()
```
Comment (dimagimon) [inline custom/_legacy/mvp/management/commands/mvp_force_update.py]: E702 multiple statements on one line (semicolon)

**Comment by dannyroberts** [comment]
(not sure about that first test failure, it doesn't happen locally, and I remember seeing and quickly fixing it in the course of working. I'll let the other tests finish before looking more)

File: corehq/apps/api/resources/v0_1.py (hunk 1 above)
Comment (benrudolph) [inline corehq/apps/api/resources/v0_1.py]: yikes

File: corehq/apps/api/resources/v0_1.py [hunk 3]
```diff
@@ -312,10 +316,15 @@ def obj_get_list(self, bundle, **kwargs):
         key = [domain]
         if case_type:
             key.append(case_type)
-        status = 'all' if closed_only else 'open'
-        cases = CommCareCase.get_all_cases(domain, case_type=case_type, status=status, include_docs=True)
-        return list(cases)
 
+        if include_closed:
```
Comment (benrudolph) [inline corehq/apps/api/resources/v0_1.py]: isn't this always `True` since it's statically defined above?

Comment (benrudolph) [inline corehq/apps/api/resources/v0_1.py]: nvm, just saw that it was being indexed in the actual file (the diff misled me)

File: custom/_legacy/mvp/management/commands/mvp_force_update.py [hunk 4]
```diff
@@ -72,20 +74,21 @@ def update_indicators_for_xmlns(self, domain, form_label_filter=None):
...
                 endkey=['submission xmlns', domain, xmlns, {}],
             ).first()
             num_forms = relevant_forms['value'] if relevant_forms else 0
-            get_forms = lambda skip, limit: XFormInstance.view("reports_forms/all_forms",
+
+            form_ids = [r['id'] for r in XFormInstance.view(
+                "reports_forms/all_forms",
                 reduce=False,
-                include_docs=True,
+                include_docs=False,
```
Comment (benrudolph) [inline custom/_legacy/mvp/management/commands/mvp_force_update.py]: :+1:

**Comment by benrudolph** [comment]
really nice cleanup. don't have too much context on what you can and cannot delete, but i think the move to dbaccessors and getting rid of `CaseQueryMixin` is sweet. test failure looks real

```
======================================================================
FAIL: testGetAllIdsOnly (corehq.apps.cloudcare.tests.test_api.CaseAPITest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/cloudcare/tests/test_api.py", line 178, in testGetAllIdsOnly
    self.assertEqual(self.expectedAll, len(list))
AssertionError: 12 != 14
```

**Comment by dannyroberts** [comment]
:white_check_mark:

**Comment by dannyroberts** [comment]
This is good to review/merge on my end

**Pull request merged**